### PR TITLE
Add support for uploading (importing) from FTP, and associated examples

### PR DESCRIPTION
**Pull request by erasche**
Covers uploading via HTTP and FTP

### CODE REVIEW

**Comment by nsoranzo** [comment]
Hi Eric,
thanks for documenting that!

The first part is fine, but I'd prefer to fix https://github.com/galaxyproject/bioblend/issues/156 instead of documenting FTP uploads with `upload1`.

**Comment by erasche** [comment]
Hi Nicola, okay, that's fine with me. 

I have some small time today, would you like me to update this PR with fixes for #156?

**Comment by erasche** [comment]
@nsoranzo added code to do what was described in #155, but I'm not sure about what needs to be done for `bioblend/galaxy/objects/wrappers.py`

File: bioblend/galaxy/tools/__init__.py [hunk 1]
```diff
@@ -138,6 +138,24 @@ def upload_file(self, path, history_id, **keywords):
         finally:
             payload["files_0|file_data"].close()
 
+    def upload_from_ftp(self, history_id, path, **keywords):
```
Comment (nsoranzo) [inline bioblend/galaxy/tools/__init__.py]: I'd swap `history_id` and `path` to be coherent with `upload_file()` and `paste_content()` methods.

File: bioblend/galaxy/tools/__init__.py [hunk 2]
```diff
@@ -138,6 +138,24 @@ def upload_file(self, path, history_id, **keywords):
...
+        :param history_id: id of the history where to upload the file
+
+        :type path: str
+        :param path: path of the file in the FTP directory
+
+        See :meth:`upload_file` for optional parameters
+        """
+        payload = self._upload_payload(history_id, **keywords)
+        payload['files_0|ftp_files'] = path
+        import pprint; pprint.pprint(payload)
```
Comment (nsoranzo) [inline bioblend/galaxy/tools/__init__.py]: Remove debugging statement.

File: bioblend/galaxy/tools/__init__.py [hunk 3]
```diff
@@ -138,6 +138,24 @@ def upload_file(self, path, history_id, **keywords):
...
+        Upload file specified by ``path`` from the user's FTP directory, to the
+        history specified by ``history_id``.
+
+        :type history_id: str
+        :param history_id: id of the history where to upload the file
+
+        :type path: str
+        :param path: path of the file in the FTP directory
+
+        See :meth:`upload_file` for optional parameters
```
Comment (nsoranzo) [inline bioblend/galaxy/tools/__init__.py]: Missing full stop at the end.

File: bioblend/galaxy/tools/__init__.py (hunk 2 above)
Comment (erasche) [inline bioblend/galaxy/tools/__init__.py]: jeez, how'd that make it in there. Whoops.

**Comment by nsoranzo** [comment]
> @nsoranzo added code to do what was described in #155, but I'm not sure about what needs to be done for bioblend/galaxy/objects/wrappers.py

Thanks a lot! I can take care of the BioBlend.objects part.

**Comment by nsoranzo** [comment]
:+1:

**Comment by jmchilton** [comment]
:+1:

**Pull request merged**